ports/stm32/powerctrlboot.c

```diff
@@ -460,13 +460,68 @@ void SystemClock_Config(void) {
 #include "stm32wlxx_ll_utils.h"
 
 void SystemClock_Config(void) {
-    // Set flash latency
+    // Set flash latency (2 wait states, sysclk > 36MHz)
     LL_FLASH_SetLatency(LL_FLASH_LATENCY_2);
     while (LL_FLASH_GetLatency() != LL_FLASH_LATENCY_2) {
     }
 
     LL_PWR_SetRegulVoltageScaling(LL_PWR_REGU_VOLTAGE_SCALE1);
 
+    #if MICROPY_HW_CLK_USE_HSE
+    // Enable the 32MHz external oscillator and 48MHZ SYSCLK via PLL
+
+    #if MICROPY_HW_CLK_USE_BYPASS
+    // Use "bypass power" option, port PB0_VDDTCXO supplies TCXO
+    // (STM32WL5x has no other HSE bypass mode.)
+
+    // "PB0 must be configured in analog mode prior enabling the HSE"
+    //
+    // Note: PB0 analog mode muxes PB0_VDDTCXO pin to the VDDTCXO regulator, set
+    // to default voltage of 1.7V.  Changing this voltage requires initializing
+    // the SUBGHZ radio and sending a Set_Tcxo command to it.
+    //
+    // For the Nucelo-WL55 board, ST uses the NDK "NT2016SF-32M-END5875A" TCXO
+    // which has no publicly available datasheet. However, the ST code for this
+    // board always keeps the pin at the default 1.7V voltage level so changing
+    // the level would only be needed if a different TCXO is used.
+    //
+    // (Note also that setting pin PB0 as a push-pull GPIO output is technically
+    // possible too, but 3.3V will be too high for many TCXOs.)
+    GPIO_InitTypeDef GPIO_InitStructure = {
+        .Pin = GPIO_PIN_0,
+        .Mode = GPIO_MODE_ANALOG,
+        .Pull = GPIO_NOPULL,
+    };
+    HAL_GPIO_Init(GPIOB, &GPIO_InitStructure);
```

Comment:
Please use `mp_hal_pin_config(pin_B0, MP_HAL_PIN_MODE_ANALOG, MP_HAL_PIN_PULL_NONE, 0)`.

Reply:
I took PB0 out of `pins.csv` in this PR because (at least in the default board configuration) it's easy to change the pin mode and disable the system clock.

Do you prefer `PB0` stays accessible from Python, or is there another way to resolve this?

Reply:
You can use `,-PB0` in `pins.csv` to make the pin accessible only from C.

Reply:
(see the comment at the top of `ports/stm32/boards/make-pins.py` for more info)

Reply:
Oh, neat! Done.